optaplanner-core/src/main/java/org/optaplanner/core/impl/io/jaxb/GenericJaxbIO.java

```diff
@@ -119,6 +119,10 @@ public T readAndValidate(Document document, String schemaResource) {
 
     private Schema readSchemaResource(String schemaResource) {
         String nonNullSchemaResource = Objects.requireNonNull(schemaResource);
+        if (GenericJaxbIO.class.getResource(nonNullSchemaResource) == null) {
```

Comment:
Getting the resource twice (once for the null check and once to pass it to the schema factory on line 138) seems wasteful to me. I'd prefer to reuse the same URL instance.